app/services/usps_daily_test_sender.rb

```diff
@@ -0,0 +1,47 @@
+# Mails a letter to the designated receiver
+class UspsDailyTestSender
+  def run
+    if valid_designated_receiver_pii?
+      UspsConfirmationMaker.new(
+        pii: designated_receiver_pii,
+        issuer: nil, # TODO: login.gov internal issuer?
+        profile: profile,
+        otp: otp_from_date,
+      ).perform
+    else
+      raise 'missing valid designated receiver pii'
+    end
+  rescue => err
+    NewRelic::Agent.notice_error(err)
+  end
+
+  # @return [String] 10-digit OTP from the date
+  # @example
+  #   "JAN20_2020"
+  def otp_from_date(date = Time.zone.today)
+    date.strftime('%b%d_%Y').upcase
+  end
+
+  # @return [Hash]
+  def designated_receiver_pii
+    @designated_receiver_pii ||= JSON.parse(
+      AppConfig.env.gpo_designated_receiver_pii,
+      symbolize_names: true,
+    )
+  end
+
+  def valid_designated_receiver_pii?
+    %i[email first_name last_name address1 city state zipcode].all? do |key|
+      designated_receiver_pii[key].present?
+    end
+  end
+
+  def profile
+    @profile ||= user.profiles.first || user.profiles.create
+  end
+
+  def user
+    email = designated_receiver_pii[:email]
+    @user ||= User.find_with_email(email) || User.create(email: email)
+  end
```

Comment:
Our DB schema doesn't allow a nil `profile_id` so it seemed easier to make up a profile than to try to relax that requirement?

My assumption is that most of the time, the designated receiver will be a login.gov team member with a valid email/user/profile, so these lines to create DB objects are more of a safety. I don't expect us to create many empty profiles/users.

Open to changing this approach if anybody feels strongly, I didn't feel super great about it

Reply:
What are the implications if the designated receiver signs in with an empty profile? Or if they try to proof? It seems like there could be some weird behavior when we try to do things like decrypt nil PII.

Could we dodge the null constraint by setting profile id to `-1` or something ugly like that?

Reply:
Updated to use profile_id = -1 in c69f1d3b1d9c98fcae958e37904bfe96f64e1934